README_rungen.md

```diff
@@ -0,0 +1,184 @@
+$ ./bin/local_laplacian.rungen --describe
+Filter name: "local_laplacian"
+  Input "input" is of type Buffer<uint16> with 3 dimensions
+  Input "levels" is of type int32
+  Input "alpha" is of type float32
+  Input "beta" is of type float32
+  Output "local_laplacian" is of type Buffer<uint16> with 3 dimensions
+```
+
+Warning: Outputs may have `$X` (where `X` is a small integer) appended to their
+names in some cases (or, in the case of Generators that don't explicitly declare
+outputs via `Output<>`, an autogenerated name of the form `fX`). If this
+happens, don't forget to escape the `$` with a backslash as necessary. These are
+both bugs we intend to fix; see https://github.com/halide/Halide/issues/2194
+
+As a convenience, there is also an implicit target that builds-and-runs, named simply "NAME.run":
+
+```
+# This is equivalent to "make local_laplacian.rungen && ./bin/local_laplacian.rungen"
+$ make local_laplacian.run
+Usage: local_laplacian.rungen argument=value [argument=value... ] [flags]
+
+# To pass arguments to local_laplacian.rungen, set the RUNARGS var:
+$ make local_laplacian.run RUNARGS=--describe
+Filter name: "local_laplacian"
+  Input "input" is of type Buffer<uint16> with 3 dimensions
+  Input "levels" is of type int32
+  Input "alpha" is of type float32
+  Input "beta" is of type float32
+  Output "local_laplacian" is of type Buffer<uint16> with 3 dimensions
+```
+
+Inputs are specified as `name=value` pairs, in any order. Scalar inputs are specified
+the typical text form, while buffer inputs (and outputs) are specified via paths to image files.
+RunGen currently can read/write image files in any format supported by halide_image_io.h; at this time, that means .png, .jpg, .ppm, .pgm, and .tmp formats. (We plan to add .tiff and .mat (level 5) in the future.)
+
+```
+$ ./bin/local_laplacian.rungen input=../apps/images/rgb16.png levels=8 alpha=1 beta=1 local_laplacian=/tmp/out.png
+$ display /tmp/out.png
+```
+
+If you specify an input or output file format that doesn't match the required
+type/dimensions for an argument (e.g., using an 8-bit PNG for an Input<float>,
+or a grayscale image for a 3-dimensional input), RunGen will try to coerce the
+inputs to something sensible; that said, it's hard to always get this right, so
+warnings are **always** issued whenever an input or output is modified in any
+way.
+
+```
+# This filter expects a 16-bit RGB image as input, but we're giving it an 8-bit grayscale image:
+$ ./bin/local_laplacian.rungen input=../apps/images/gray.png levels=8 alpha=1 beta=1 local_laplacian=/tmp/out.png
+Warning: Image for Input "input" has 2 dimensions, but this argument requires at least 3 dimensions: adding dummy dimensions of extent 1.
+Warning: Image loaded for argument "input" is type uint8 but this argument expects type uint16; data loss may have occurred.
+```
+
+By default, we try to guess a suitable size for the output image(s), based on
```

Comment:
I don't think bounds query logic can do this. Doesn't rungen just use the input size if the output size is not specified?

Reply:
It guesses based on the input size, yes. The reference to bounds-query logic is misleading: there is a comment in RunGen.cpp suggesting that using the algorithm from AddImageChecks would probably produce a better guess. I'l reword the text.